src/executables/util_cs.c

```diff
@@ -3260,53 +3271,64 @@ applyinfo (UTIL_FUNCTION_ARG * arg)
-
-	  if (check_database_name (local_database_name))
+	  if ((check_applied_info != false) && check_database_name (local_database_name))
 	    {
-	      goto check_applied_info_end;
+	      printf ("\n *** Applied Info. *** \n");
+	      check_applied_info = false;
+	      PRINT_AND_LOG_ERR_MSG ("\nERROR : %s\n", db_error_string (3));
 	    }
-	  if (db_login ("DBA", NULL) != NO_ERROR)
+	  if ((check_applied_info != false) && (db_login ("DBA", NULL) != NO_ERROR))
 	    {
-	      goto check_applied_info_end;
+	      printf ("\n *** Applied Info. *** \n");
+	      check_applied_info = false;
+	      PRINT_AND_LOG_ERR_MSG ("\nERROR : %s\n", db_error_string (3));
 	    }
-	  error = db_restart (arg->command_name, TRUE, local_database_name);
-	  if (error != NO_ERROR)
+	  if ((check_applied_info != false) && (db_restart (arg->command_name, TRUE, local_database_name) != NO_ERROR))
 	    {
-	      goto check_applied_info_end;
+	      printf ("\n *** Applied Info. *** \n");
+	      check_applied_info = false;
+	      PRINT_AND_LOG_ERR_MSG ("\nERROR : %s\n", db_error_string (3));
 	    }
 
-	  if (HA_DISABLED ())
+	  if ((check_applied_info != false) && HA_DISABLED ())
 	    {
+	      printf ("\n *** Applied Info. *** \n");
+	      check_applied_info = false;
+	      printf ("\nERROR : ");
 	      PRINT_AND_LOG_ERR_MSG (msgcat_message (MSGCAT_CATALOG_UTILS, MSGCAT_UTIL_SET_APPLYINFO,
 						     APPLYINFO_MSG_NOT_HA_MODE));
-	      goto check_applied_info_end;
 	    }
+	  error = NO_ERROR;
 
-	  error = la_log_page_check (local_database_name, log_path, pageid, check_applied_info, check_copied_info,
-				     check_replica_info, verbose, &copied_eof_lsa, &copied_append_lsa,
+	  error = la_log_page_check (local_database_name, log_path, pageid, &check_applied_info, &check_copied_info,
+				     &check_replica_info, verbose, &copied_eof_lsa, &copied_append_lsa,
 				     &applied_final_lsa);
-	  (void) db_shutdown ();
+	  if (error == NO_ERROR)
+	    {
+	      (void) db_shutdown ();
+	    }
 	}
       else if (check_copied_info)
 	{
 	  memset (local_database_name, 0x00, CUB_MAXHOSTNAMELEN);
 	  strcpy (local_database_name, database_name);
 	  strcat (local_database_name, "@localhost");
 
-	  error = la_log_page_check (local_database_name, log_path, pageid, check_applied_info, check_copied_info,
-				     check_replica_info, verbose, &copied_eof_lsa, &copied_append_lsa,
+	  error = la_log_page_check (local_database_name, log_path, pageid, &check_applied_info, &check_copied_info,
+				     &check_replica_info, verbose, &copied_eof_lsa, &copied_append_lsa,
 				     &applied_final_lsa);
 	}
 
-    check_applied_info_end:
       if (error != NO_ERROR)
 	{
-	  PRINT_AND_LOG_ERR_MSG ("%s\n", db_error_string (3));
+	  PRINT_AND_LOG_ERR_MSG ("\nERROR : %s\n", db_error_string (3));
+	  check_applied_info = false;
+	  (void) db_shutdown ();
 	}
       error = NO_ERROR;
-
       if (check_master_info)
 	{
+	  printf ("\n ***  Active Info. *** \n");
```

Comment:
It looks to be printed twice when `logwr_copy_log_header_check()` succeeds.

Reply:
I identified. And only scalability version code has `*** Active Info. ***` message in the `logwr_copy_log_header_check()`.

Reply:
> I identified. 

What do you mean? You mean that you already know what I mentioned?

> And only scalability version code has `*** Active Info. ***` message in the `logwr_copy_log_header_check()`.

Do you mean the scalability_dev branch? `logwr_copy_log_header_check()` seems to print the message always when it succeeds.

Reply:
hahaha It is my mistake. I already eliminated this `printf ("\n *** Active Info. *** \n");` So I was confused it.

Can we see the code on `src/transaction/log_writer.c` in this `file changed` tab?
(I cannot link but there is the code bottom of this page)

Reply:
> What do you mean? You mean that you already know what I mentioned?

I wrote 'I identified' when I see my already changed code.